server/routerlicious/packages/routerlicious-base/src/alfred/services/deltaService.ts

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { toUtf8 } from "@fluidframework/common-utils";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { ITenantManager, MongoManager } from "@fluidframework/server-services-core";
+
+export class DeltaService {
+    constructor(
+        protected readonly mongoManager: MongoManager,
+        protected readonly tenantManager: ITenantManager,
+    ) {
+    }
+
+    public async getDeltas(
+        collectionName: string,
+        tenantId: string,
+        documentId: string,
+        from?: number,
+        to?: number): Promise<ISequencedDocumentMessage[]> {
+        // Create an optional filter to restrict the delta range
+        const query: any = { documentId, tenantId };
+        if (from !== undefined || to !== undefined) {
+            query["operation.sequenceNumber"] = {};
+
+            if (from !== undefined) {
+                query["operation.sequenceNumber"].$gt = from;
+            }
+
+            if (to !== undefined) {
+                query["operation.sequenceNumber"].$lt = to;
+            }
+        }
+
+        const sort = { "operation.sequenceNumber": 1 };
+        return this.queryDeltas(collectionName, query, sort);
+    }
+
+    public async getDeltasFromStorage(
+        collectionName: string,
+        tenantId: string,
+        documentId: string,
+        fromTerm: number,
+        toTerm: number,
+        fromSeq?: number,
+        toSeq?: number): Promise<ISequencedDocumentMessage[]> {
+        const query: any = { documentId, tenantId, scheduledDeletionTime: { $exists: false } };
+        query["operation.term"] = {};
+        query["operation.sequenceNumber"] = {};
+        query["operation.term"].$gte = fromTerm;
+        query["operation.term"].$lte = toTerm;
+        if (fromSeq !== undefined) {
+            query["operation.sequenceNumber"].$gt = fromSeq;
+        }
+        if (toSeq !== undefined) {
+            query["operation.sequenceNumber"].$lt = toSeq;
+        }
+
+        const sort = { "operation.term": 1, "operation.sequenceNumber": 1 };
+        return this.queryDeltas(collectionName, query, sort);
+    }
+
+    private async queryDeltas(collectionName: string, query: any, sort: any): Promise<ISequencedDocumentMessage[]> {
+        const db = await this.mongoManager.getDatabase();
+        // eslint-disable-next-line @typescript-eslint/await-thenable
+        const collection = await db.collection<any>(collectionName);
```

Comment:
`db.collection` is NOT awaitable. Not sure why we want to await and put a disable rule `eslint-disable-next-line @typescript-eslint/await-thenable`. Any objections that I take out the `await`?

Reply:
OK, I just removed the `await`.